docs/metadata-enumeration.md

```diff
@@ -0,0 +1,27 @@
+---
+title: Metadata Enumeration
+---
+
+Connection metadata refers to the set of APIs provided by the Connectivity team to model Warehouse/Database/Schema/Table hierarchy, as well as table column details like name and type in the relational data sources. When connecting to a data source Tableau needs to know the hierarchy, and enumerate entities. When selecting a table or connecting to an existing saved connection the column level info needs to be queried as well. 
+
+## ODBC Metadata Enumeration
+ODBC capabilities can determine the actual SQL query generated by Tableau to read ODBC metadata, there are several scenarios listed below. Note that those scenarios are in order, Whatever is met first will not reach the following scenarios. The detailed logs of reading ODBC metadata are logged with `SQLODBCProtocol::ReadMetadataImpl` keyword, you should be able to find those log lines in tabprotosrv log file.
+
+1. Full "select *" using SQLPrepare only, which ensures the query is parsed but not executed.
+Conditions: CAP_ODBC_METADATA_SUPPRESS_PREPARED_QUERY = false AND CAP_ODBC_METADATA_SUPRESS_SELECT_STAR = false
+Example query: `SELECT * FROM default.testv1_batters`
+
+2. Row-limiting "select *" with WHERE 1=0.
+Conditions: CAP_QUERY_WHERE_FALSE_METADATA = true AND (CAP_ODBC_METADATA_SUPPRESS_PREPARED_QUERY = false OR CAP_ODBC_METADATA_SUPPRESS_EXECUTED_QUERY = false)
+Example query: `SELECT * FROM default.testv1_batters WHERE 1=0`
+
+3. Row-limiting "select *" with TOP 0 / LIMIT 0.
+Conditions: CAP_QUERY_TOP_0_METADATA = true AND (CAP_ODBC_METADATA_SUPPRESS_PREPARED_QUERY = false OR CAP_ODBC_METADATA_SUPPRESS_EXECUTED_QUERY = false)
+Example query: `SELECT * FROM default.testv1_batters LIMIT 0`
+
+4. Less reliable but fast API for querying table metadata directly from [SQLColumns](https://docs.microsoft.com/en-us/sql/odbc/reference/syntax/sqlcolumns-function?view=sql-server-ver15) calls
+Conditions: CAP_ODBC_METADATA_SUPPRESS_SQLCOLUMNS_API = false
+
+5. Full "select *" query
+Conditions: CAP_ODBC_METADATA_SUPPRESS_PREPARED_QUERY = false OR CAP_ODBC_METADATA_SUPPRESS_EXECUTED_QUERY = false
+Example query: `SELECT * FROM default.testv1_batters`
```

Comment:
Can you add an additional section that links to the capabilities doc as additional reference.  Some of those CAP_ use slightly different wording, but I think this as written is the best high level summary to start with.